services/gateway/tests/unit/registerRpcApi.test.js

```diff
@@ -0,0 +1,160 @@
+		expect(() => curlyBracketsToColon(null)).toThrow();
+	});
+
+	it('should throw when called with undefined', async () => {
+		expect(() => curlyBracketsToColon(undefined)).toThrow();
+	});
+});
+
+describe('Test convertType method', () => {
+	it('should return number when called to convert string to number', async () => {
+		const response = convertType('123', 'number');
+		expect(typeof response).toEqual('number');
+		expect(response).toEqual(123);
+	});
+
+	it('should return string when called to convert number to string', async () => {
+		const response = convertType(123, 'string');
+		expect(typeof response).toEqual('string');
+		expect(response).toEqual('123');
+	});
+
+	it('should return string when called to convert array to string', async () => {
+		const response = convertType(new Array(...[1, 2, 3]), 'string');
+		expect(typeof response).toEqual('string');
+		expect(response).toEqual('1,2,3');
+	});
+
+	it('should return boolean when called to convert string to boolean', async () => {
+		const response = convertType('true', 'boolean');
+		expect(typeof response).toEqual('boolean');
+		expect(response).toEqual(true);
+	});
+
+	it('should return param type when called for unknown conversion', async () => {
+		const response = convertType(true, 'object');
+		expect(typeof response).toEqual('boolean');
+		expect(response).toEqual(true);
+	});
+});
+
+describe('Test mapParam method', () => {
+	const source = {
+		originalKey: 'originalValue',
+		mappingKey: 'mappingValue',
+	};
+
+	it('should return value of mapping key when originalKey is `=`', async () => {
+		const response = mapParam(source, '=', 'mappingKey');
+		expect(response).toEqual({
+			key: 'mappingKey',
+			value: 'mappingValue',
+		});
+	});
+
+	it('should return value of original key when originalKey is not `=`', async () => {
+		const response = mapParam(source, 'originalKey', 'mappingKey');
+		expect(response).toEqual({
+			key: 'mappingKey',
+			value: 'originalValue',
+		});
+	});
+
+	it('should return empty object when mappingKey is null or undefined', async () => {
+		// Null
+		const response = mapParam(source, 'originalKey', null);
+		expect(response).toEqual({});
+
+		// Undefined
+		const response2 = mapParam(source, 'originalKey', null);
+		expect(response2).toEqual({});
+	});
+
+	it('should return undefined value when originalKey is null or undefined', async () => {
+		// Null
+		const response = mapParam(source, null, 'mappingKey');
+		expect(response).toEqual({
+			key: 'mappingKey',
+			value: undefined,
+		});
+
+		// Undefined
+		const response2 = mapParam(source, undefined, 'mappingKey');
+		expect(response2).toEqual({
+			key: 'mappingKey',
+			value: undefined,
+		});
+	});
+
+	it('should throw error when source is null or undefined', async () => {
+		// Null
+		expect(() => mapParam(null, 'originalKey', 'mappingKey')).toThrow();
+
+		// Undefined
+		expect(() => mapParam(null, 'originalKey', 'mappingKey')).toThrow();
```

Comment:
```suggestion
		expect(() => mapParam(undefined, 'originalKey', 'mappingKey')).toThrow();
```
or this can be replaced with: 
```suggestion
		it('should throw error when source is null or undefined', async () => {
		[null, undefined].forEach(entry => {
			expect(() => mapParam(entry, 'originalKey', 'mappingKey')).toThrow()
		})
	});
```

Reply:
Applied.